Snowflake.Data.Tests/UnitTests/SnowflakeDbConnectionTest.cs

```diff
@@ -0,0 +1,61 @@
+
+
+using System;
+using System.IO;
+using Mono.Unix;
+
+namespace Snowflake.Data.Tests.UnitTests
+{
+    using Core;
+    using Core.Tools;
+    using Moq;
+    using NUnit.Framework;
+    using Snowflake.Data.Client;
+
+    public class SnowflakeDbConnectionTest
+    {
+        [Test]
+        public void TestFillConnectionStringFromTomlConfig()
+        {
+            // Arrange
+            var mockFileOperations = new Mock<FileOperations>();
+            var mockEnvironmentOperations = new Mock<EnvironmentOperations>();
+            mockEnvironmentOperations.Setup(e => e.GetFolderPath(Environment.SpecialFolder.UserProfile))
+                .Returns($"{Path.DirectorySeparatorChar}home");
+            mockFileOperations.Setup(f => f.Exists(It.IsAny<string>())).Returns(true);
+            mockFileOperations.Setup(f => f.ReadAllText(It.IsAny<string>(), It.IsAny<Action<UnixStream>>()))
+                .Returns("[default]\naccount=\"testaccount\"\nuser=\"testuser\"\npassword=\"testpassword\"\n");
+            var tomlConnectionBuilder = new SnowflakeTomlConnectionBuilder(mockFileOperations.Object, mockEnvironmentOperations.Object);
+
+            // Act
+            using (var conn = new SnowflakeDbConnection(tomlConnectionBuilder))
+            {
+                conn.FillConnectionStringFromTomlConfigIfNotSet();
+                // Assert
+                Assert.AreEqual("account=testaccount;user=testuser;password=testpassword;", conn.ConnectionString);
+            }
+        }
+
+        [Test]
+        public void TestFillConnectionStringFromTomlConfigShouldNotBeExecutedIfAlreadySetConnectionString()
```

Comment:
the test name is not good, because the method `Fill...` will always be executed. The thing is that it won't modify connection string. So maybe `TestTomlConfigurationDoesNotOverrideExistingConnectionString` ?

Reply:
Done